src/libcore/yk_swt_impl.c

```diff
@@ -114,12 +110,15 @@ yk_swt_stop_tracing_impl(size_t *ret_trace_len) {
     trace_buf = NULL;
     tracing = false;
     trace_buf_len = 0;
+    trace_buf_cap = 0;
 
     return ret_trace;
 }
 
 // Call this to safely mark the trace invalid.
+// We don't free the trace buffer here, as this may be called in a signal handler
```

Comment:
The second sentence should be inside the function IMHO: it's not something that affects the caller.

Reply:
Agreed.